File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/ResetFrequency.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ResetFrequency : Frequency
    {
        public ResetFrequency(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode weeklyRollConventionNode = xmlNode.SelectSingleNode("weeklyRollConvention");
            
            if (weeklyRollConventionNode != null)
            {
                if (weeklyRollConventionNode.Attributes["href"] != null || weeklyRollConventionNode.Attributes["id"] != null) 
                {
                    if (weeklyRollConventionNode.Attributes["id"] != null) 
                    {
                        weeklyRollConventionIDRef_ = weeklyRollConventionNode.Attributes["id"].Value;
                        WeeklyRollConventionEnum ob = new WeeklyRollConventionEnum(weeklyRollConventionNode);
                        IDManager.SetID(weeklyRollConventionIDRef_, ob);
                    }
                    else if (weeklyRollConventionNode.Attributes["href"] != null)
                    {
                        weeklyRollConventionIDRef_ = weeklyRollConventionNode.Attributes["href"].Value;
                    }
                    else
                    {
                        weeklyRollConvention_ = new WeeklyRollConventionEnum(weeklyRollConventionNode);
                    }
                }
                else
                {
                    weeklyRollConvention_ = new WeeklyRollConventionEnum(weeklyRollConventionNode);
                }
            }
            
        
        }
        
    
        #region WeeklyRollConvention_
        private WeeklyRollConventionEnum weeklyRollConvention_;
        public WeeklyRollConventionEnum WeeklyRollConvention_
        {
            get
            {
                if (this.weeklyRollConvention_ != null)
                {
                    return this.weeklyRollConvention_; 
                }
                else if (this.weeklyRollConventionIDRef_ != null)
                {
                    weeklyRollConvention_ = IDManager.getID(weeklyRollConventionIDRef_) as WeeklyRollConventionEnum;
                    return this.weeklyRollConvention_; 
                }
                else
                {
                      return this.weeklyRollConvention_; 
                }
            }
            set
            {
                if (this.weeklyRollConvention_ != value)
                {
                    this.weeklyRollConvention_ = value;
                }
            }
        }
        #endregion
        
        public string weeklyRollConventionIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/ResetFrequency.cpp
// ResetFrequency.cpp 
#include "ResetFrequency.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ResetFrequency::ResetFrequency(TiXmlNode* xmlNode)
: Frequency(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //weeklyRollConventionNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* weeklyRollConventionNode = xmlNode->FirstChildElement("weeklyRollConvention");

   if(weeklyRollConventionNode){weeklyRollConventionIsNull_ = false;}
   else{weeklyRollConventionIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- weeklyRollConventionNode , address : " << weeklyRollConventionNode << std::endl;
   #endif
   if(weeklyRollConventionNode)
   {
      if(weeklyRollConventionNode->Attribute("href") || weeklyRollConventionNode->Attribute("id"))
      {
          if(weeklyRollConventionNode->Attribute("id"))
          {
             weeklyRollConventionIDRef_ = weeklyRollConventionNode->Attribute("id");
             weeklyRollConvention_ = boost::shared_ptr<WeeklyRollConventionEnum>(new WeeklyRollConventionEnum(weeklyRollConventionNode));
             weeklyRollConvention_->setID(weeklyRollConventionIDRef_);
             IDManager::instance().setID(weeklyRollConventionIDRef_,weeklyRollConvention_);
          }
          else if(weeklyRollConventionNode->Attribute("href")) { weeklyRollConventionIDRef_ = weeklyRollConventionNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { weeklyRollConvention_ = boost::shared_ptr<WeeklyRollConventionEnum>(new WeeklyRollConventionEnum(weeklyRollConventionNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<WeeklyRollConventionEnum> ResetFrequency::getWeeklyRollConvention()
{
   if(!this->weeklyRollConventionIsNull_){
        if(weeklyRollConventionIDRef_ != ""){
             return boost::shared_static_cast<WeeklyRollConventionEnum>(IDManager::instance().getID(weeklyRollConventionIDRef_));
        }else{
             return this->weeklyRollConvention_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<WeeklyRollConventionEnum>();
   }
}
}

